app/src/main/java/org/oppia/app/application/OppiaApplication.kt

```diff
@@ -10,22 +10,24 @@ import org.oppia.domain.oppialogger.exceptions.OppiaUncaughtExceptionHandler
 /** The root [Application] of the Oppia app. */
 class OppiaApplication : MultiDexApplication(), ActivityComponentFactory {
   /** The root [ApplicationComponent]. */
-
-  lateinit var oppiaUncaughtExceptionHandler: OppiaUncaughtExceptionHandler
-
   private val component: ApplicationComponent by lazy {
     DaggerApplicationComponent.builder()
       .setApplication(this)
       .build()
   }
 
+  private lateinit var oppiaUncaughtExceptionHandler: OppiaUncaughtExceptionHandler
```

Comment:
I'm still not keen on changing the OppiaApplication class, especially since it's going to make later migrations (such as the one to Dagger Hilt) more difficult. The current solution also won't correctly integrate with test applications which will make testing in the app module more difficult. See this gist for an idea on how to solve this using Dagger listeners as I suggested earlier:

https://gist.github.com/BenHenning/93efec191657b72667a476c456a0de5f

Reply:
Okay, I got your point and upon thinking I also feel that it'll be a lot better if we do things this way. Thanks a lot for the gist. It really helped me understand the setup.